src/instructlab/sdg/utils/docprocessor.py

```diff
@@ -0,0 +1,637 @@
+        server_ctx_size (int): Context window size of server.
+        chunk_word_count (int): Maximum number of words to chunk a document.
+    Returns:
+         List[str]: List of chunked documents.
+    """
+    num_tokens_per_doc = _num_tokens_from_words(chunk_word_count)
+    if num_tokens_per_doc > int(server_ctx_size - 1024):
+        raise ValueError(
+            "Error: {}".format(
+                str(
+                    f"Given word count ({chunk_word_count}) per doc will exceed the server context window size ({server_ctx_size})"
+                )
+            )
+        )
+    if documents == []:
+        return []
+
+    # Placeholder for params
+    content = []
+    chunk_size = _num_chars_from_tokens(num_tokens_per_doc)
+    chunk_overlap = _DEFAULT_CHUNK_OVERLAP
+
+    # Using Markdown as default, document-specific chunking will be implemented in separate pr.
+    md_text_splitter = RecursiveCharacterTextSplitter.from_language(
+        language=Language.MARKDOWN,
+        chunk_size=chunk_size,
+        chunk_overlap=chunk_overlap,
+    )
+
+    # Determine file type for heuristics, default with markdown
+    for doc in documents:
+        # Use regex to remove unnecessary dashes in front of pipe characters in a markdown table.
+        doc = re.sub(r"-{2,}\|", "-|", doc)
+        # Remove unnecessary spaces in front of pipe characters in a markdown table.
+        doc = re.sub(r"\  +\|", " |", doc)
+        temp = md_text_splitter.create_documents([doc])
+        content.extend([item.page_content for item in temp])
+    return content
+
+
+def chunk_pdfs(pdf_docs: List, filepaths: List, leaf_node_path: Path, model_name: str = None):
+    """Semantically chunk PDF documents.
+
+    TODO
+    """
+    if pdf_docs == []:
+        return []
+
+    print(f"""THIS IS KHALED: CHUNKING PDF DOCS
+        {pdf_docs[0]=}
+    """)
+    artifacts_path = DocumentConverter.download_models_hf()
+    converter = DocumentConverter(artifacts_path=artifacts_path)
+    inputs = DocumentConversionInput.from_paths(filepaths)  
+    parsed_pdfs = converter.convert(inputs)
+    print(f"THIS IS KHALED: {parsed_pdfs=}")
+
+    docling_jsons_path = DOC_FILEPATH / "docling-jsons"
+    docling_jsons_path.mkdir(parents=True, exist_ok=True)
+
+    export_documents(parsed_pdfs, docling_jsons_path)
+    parsed_dicts = [p.render_as_dict() for p in parsed_pdfs]
+
+
+    # TODO name files better
+    for i, pd in enumerate(parsed_dicts):
+        fp = docling_jsons_path / f"docling_{i}.json"
+
+        with open(fp, "w") as json_file:
+            for entry in pd:
+                json_file.write(json.dumps(entry) + "\n")
+
+    dp = DocProcessor(
+        parsed_doc_dir=str(docling_jsons_path),
+        tokenizer_model_name=model_name,
+        qna_yaml_path=Path("~/.local/share/instructlab/taxonomy").expanduser() / leaf_node_path / "qna.yaml",
+    )
+
+    chunked_pdfs = list(dp.get_processed_dataset())
```

Comment:
@bbrowning we are calling `get_processed_dataset` here.

Reply:
Ahh, yes. That was added between my comment above and this comment, so marking the comment above as resolved :smile: